src/crud_ops.rs

```diff
@@ -253,3 +378,19 @@ export_queued_op!(
     },
     path: *const c_char
 );
+
+export_queued_op!(
+    bulk_delete,
+    BulkResponse,
+    |config, response| {
+        let mut paths_vec:Vec<Path> = Vec::new();
+        for i in 0..num_paths as isize {
+            let path_ptr = unsafe { *path_c_array.offset(i)};
+            let path = unsafe { std::ffi::CStr::from_ptr(path_ptr) };
+            let path = unsafe { cstr_to_path(path) };
+            paths_vec.push(path);
+        }
```

Comment:
Using `offset` here is correct but it is more common in Rust FFI code to always construct a slice for C arrays like:
```suggestion
        let slice: &[*const c_char] = unsafe { std::slice::from_raw_parts(path_c_array, num_paths) };
        let paths_vec: Vec<Path> = slice.iter().map(|ptr| {
            unsafe { cstr_to_path(std::ffi::CStr::from_ptr(ptr)) }
        }).collect();
```